### DISCUSSION TITLE
Reach WalletConnect fails to sign transactions in React Hooks, promise waits forever

### DISCUSSION THREAD

**Discussion by AtegyekaRodgers** [reach-sh/reach-lang#948] [Q&A] [answered]
![algo_signTxn](https://user-images.githubusercontent.com/47475700/167692857-86809958-1aad-4f27-a855-26a42936e435.png)
By monitoring the logs in my browser console, I can see that WalletConnect gets stuck at the step waiting for a transaction to be signed. 
I would like to know how I can have the transactions signed explicitly so that my app does not halt.

My code repository is this: https://github.com/Reach-Insurance/react

Specifically line `271 ` inside a function named  `deployContract`
```js
  const deployContract = async (e) => {
  //...
  //...
  console.log("Setting the initial state of the contract just deployed");
    await insurerContract.current.p.Insurer(interact.current);
   
    console.log("getting the contract info ...");
    const info = await insurerContract.current.getInfo();
    console.log("info =", info);
    const infoStr = JSON.stringify(info);
    
    //...
    //...
    }
    ```

**Comment by AtegyekaRodgers** [comment, marked as answer]
The promise on line  271 is `await`ed forever. It never resolves

**Reply by hmzdot** [reply]
Participant functions resolve when contract is destroyed. If you don't intend this you should run it without await to keep it in the background

```js
insurerContract.current.p.Insurer(interact.current);
```

**Reply by AtegyekaRodgers** [reply]
Ok, makes sense. Let me try that
